pages/blog/the-reference-rabbit-hole.md

```diff
@@ -0,0 +1,275 @@
+
+## What about `$schema`?
+In JSON Schema Draft 7, and in the **Schema Object**, there exist a keyword, similar to what `schemaFormat` is for the AsyncAPI Message Object, that can be used to define what version of JSON Schema `LightMeasurement` follows.
+
+So what if both are defined at the same time?
+
+```yaml
+asyncapi: '2.2.0'
+...
+components:
+  messages:
+    LightMeasured: 
+      name: LightMeasured
+      schemaFormat: application/vnd.aai.asyncapi+yaml;version=2.2.0
+      payload:
+        $ref: '#/components/schemas/LightMeasurement'
+  schemas:
+    LightMeasurement:
+      $schema: 'http://json-schema.org/draft-04/schema#'
+      ...
+```
+With such contradicting information, how should tooling handle this? This sparked [spec #655](https://github.com/asyncapi/spec/issues/655).
+
+## What about extra keywords?
+Following that, by taking a closer look at the [JSON reference](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03) specification the **Reference Object** follows, we find the [sentence](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03#section-3):
+
+> Any members other than "$ref" in a JSON Reference object SHALL be ignored.
+
+What this means, is that if we have a reference defined such as:
+
+```yaml
+...
+components:
+  messages:
+    LightMeasured:
+      payload:
+        type: object
+        $ref: '#/components/schemas/LightMeasurement'
+```
+
+
+The `type` property for the message payload, should be completely ignored. So let's try and see what happens when we try this in [Studio](https://studio.asyncapi.com/?base64=YXN5bmNhcGk6ICcyLjIuMCcKaW5mbzoKICB0aXRsZTogVGVzdCBvdmVycmlkaW5nIHByb3BlcnRpZXMgd2l0aCBkZXJlZmVyZW5jZWQgb2JqZWN0cyAKICB2ZXJzaW9uOiAnMS4wLjAnCmNoYW5uZWxzOgogIHRlc3Q6CiAgICBwdWJsaXNoOgogICAgICBtZXNzYWdlOgogICAgICAgICRyZWY6ICcjL2NvbXBvbmVudHMvbWVzc2FnZXMvTGlnaHRNZWFzdXJlbWVudCcKY29tcG9uZW50czoKICBtZXNzYWdlczoKICAgIExpZ2h0TWVhc3VyZW1lbnQ6IAogICAgICBuYW1lOiBMaWdodE1lYXN1cmVtZW50CiAgICAgIHBheWxvYWQ6CiAgICAgICAgdHlwZTogYm9vbGVhbgogICAgICAgICRyZWY6ICcjL2NvbXBvbmVudHMvc2NoZW1hcy9MaWdodE1lYXN1cmVtZW50JwogIHNjaGVtYXM6CiAgICBMaWdodE1lYXN1cmVtZW50OgogICAgICB0eXBlOiBzdHJpbmc=).
```

Comment:
I wouldn't force the user to go to the studio to see what happens. I recommend you to mention at least that the property `type` is wrongly being overriding or something like that.

Reply:
Good point, changed